cms/home/templatetags/url_translations.py

```diff
@@ -0,0 +1,30 @@
+from django import template
+from django.core.urlresolvers import resolve, reverse
+from django.http import Http404
+from django.utils import translation
+
+register = template.Library()
+
+
+@register.simple_tag(takes_context=True)
+def translate_url(context, lang=None, *args, **kwargs):
+    """
+    Get active page's url by a specified language
+    Usage: {% translate_url 'en' %}
+    """
+    path = context['request'].path
+    cur_language = translation.get_language()
+
+    try:
+        view = resolve(path)
+        translation.activate(lang)
+        url = reverse(
+            view.view_name,
+            args=view.args,
+            kwargs=view.kwargs,
+        )
+    except Http404:
+        url = '/' + lang + '/'
+    finally:
```

Comment:
You can leave finally out, by using the [build in context manager](https://docs.djangoproject.com/en/1.11/ref/utils/#django.utils.translation.override).